### ISSUE TITLE
grub2: unreproducible localization resources

### ISSUE THREAD

**Issue by raboof** [NixOS/nixpkgs#245356]
Building this package twice does not produce the bit-by-bit identical result each time, making it harder to detect CI breaches. You can read more about this at https://reproducible-builds.org/ .

Fixing bit-by-bit reproducibility also has additional advantages, such as avoiding hard-to-reproduce bugs, making content-addressed storage more effective and reducing rebuilds in such systems.

### Steps To Reproduce

```
nix-build '<nixpkgs>' -A grub2 --check --keep-failed
```

You can use `diffoscope` to analyze the differences in the output of the two builds.

To view the build log of the build that produced the artifact in the binary cache:

```
nix-store --read-log $(nix-instantiate '<nixpkgs>' -A grub2)
```

### Additional context

https://reproducible.nixos.org/nixos-iso-minimal-runtime/diff/76610f85a7c73b54e1ec6fdee1982929a5391d7122f590d0cc7eeae4fee9f5c9-9ca84891f8328dd999d6e6373b10cb8018ab97814d9c28f00148901d5137577a.html

possibly related: https://savannah.gnu.org/bugs/?59658

**Comment by delroth** [comment]
I think we haven't hit this problem before (with 2.06) because the `po` files are usually distributed along with the source tarball. Now (with 2.12-rc1) that we've switched to a git checkout the derivation is building the `po` files itself, and I suspect we're either not doing it right, or we're missing part of the process.

**Comment by delroth** [comment]
(Fixing gettext's reproducibility is also likely an acceptable solution - but I suspect there's a more targeted and less controversial fix we could just apply to grub2 directly.)

**Comment by raboof** [comment]
should be fixed with #245885 on staging - if we can do something to grub2 to fix it 'more targeted and less controversial' that's still fine, but might not be needed
